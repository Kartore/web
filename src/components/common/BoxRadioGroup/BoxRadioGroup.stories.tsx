import type { Meta, StoryObj } from '@storybook/react';

import { BoxRadioGroup } from '.';

const meta = {
  component: BoxRadioGroup,
} satisfies Meta<typeof BoxRadioGroup>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    label: 'Numbers',
    items: [
      { value: '1', label: 'One' },
      { value: '2', label: 'Two' },
      { value: '3', label: 'Three' },
    ],
  },
};
