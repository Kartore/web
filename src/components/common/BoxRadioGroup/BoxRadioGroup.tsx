import type { FC } from 'react';

import type { AriaRadioGroupProps } from 'react-aria';
import { useRadioGroup } from 'react-aria';
import { useRadioGroupState } from 'react-stately';

import { BoxRadio } from '~/components/common/BoxRadioGroup/BoxRadio';
import { cn } from '~/utils/tailwindUtil';

export type BoxRadioGroupProps = {
	className?: string;
	items: { value: string; label: string }[];
} & AriaRadioGroupProps;

export const BoxRadioGroup: FC<BoxRadioGroupProps> = ({
	className,
	label,
	items,
	...props
}) => {
	const state = useRadioGroupState({ label, ...props });
	const { radioGroupProps, labelProps } = useRadioGroup(
		{ label, ...props },
		state,
	);
	return (
		<div
			{...radioGroupProps}
			className={cn('flex items-center justify-between text-sm', className)}
		>
			<span
				{...labelProps}
				className={cn('font-semibold text-gray-600', labelProps.className)}
			>
				{label}
			</span>
			<div className={'flex w-1/2 flex-row items-center justify-end gap-1'}>
				{items.map((item) => {
					return (
						<BoxRadio state={state} value={item.value} key={item.value}>
							{item.label}
						</BoxRadio>
					);
				})}
			</div>
		</div>
	);
};

BoxRadioGroup.displayName = 'BoxRadioGroup';
