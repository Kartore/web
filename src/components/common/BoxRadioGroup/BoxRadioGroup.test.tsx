import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { describe } from 'vitest';

import * as Stories from './BoxRadioGroup.stories';

const { Default } = composeStories(Stories);

describe('Component: BoxRadioGroup', () => {
	describe('Snapshot', () => {
		it('Default', () => {
			const { asFragment } = render(<Default />);
			expect(asFragment()).toMatchSnapshot();
		});
	});
});
