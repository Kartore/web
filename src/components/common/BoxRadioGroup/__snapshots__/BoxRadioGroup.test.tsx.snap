// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: BoxRadioGroup > Snapshot > Default 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="react-aria-:r1:"
    aria-orientation="vertical"
    class="flex items-center justify-between text-sm"
    id="react-aria-:r0:"
    role="radiogroup"
  >
    <span
      class="font-semibold text-gray-600"
      id="react-aria-:r1:"
    >
      Numbers
    </span>
    <div
      class="flex w-1/2 flex-row items-center justify-end gap-1"
    >
      <label
        aria-checked="false"
        class="block cursor-pointer rounded border-none bg-gray-100 py-1 px-2 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0 aria-checked:bg-gray-300"
      >
        <input
          class="sr-only"
          name="react-aria-:r5:"
          tabindex="0"
          type="radio"
          value="1"
        />
        One
      </label>
      <label
        aria-checked="false"
        class="block cursor-pointer rounded border-none bg-gray-100 py-1 px-2 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0 aria-checked:bg-gray-300"
      >
        <input
          class="sr-only"
          name="react-aria-:r5:"
          tabindex="0"
          type="radio"
          value="2"
        />
        Two
      </label>
      <label
        aria-checked="false"
        class="block cursor-pointer rounded border-none bg-gray-100 py-1 px-2 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0 aria-checked:bg-gray-300"
      >
        <input
          class="sr-only"
          name="react-aria-:r5:"
          tabindex="0"
          type="radio"
          value="3"
        />
        Three
      </label>
    </div>
  </div>
</DocumentFragment>
`;
